### ISSUE TITLE
--weights_path ./weight/X-152.pth where download?

### ISSUE THREAD

**Issue by alice-cool** [rentainhe/TRAR-VQA#2]
Dear scholar, I want to extract the 16 times 16 features using the ResNext152 described in your paper.

**Comment by alice-cool** [comment]
https://github.com/facebookresearch/grid-feats-vqa  is it right?

**Comment by alice-cool** [comment]
When I INPUT "CUDA_VISIBLE_DEVICES=1 python extract_trar_grid_feature.py --config-file configs/X-152-grid.yaml --dataset coco_2015_test --weight_path ./weight/X-152.pth --output_dir ./data/"   to extract from coco test2015.zip jpg.
It throws the exception like the following:

 **raise KeyError(
KeyError: "**Dataset 'coco_2015_test' is not registered!

![image](https://user-images.githubusercontent.com/16870890/192797994-d97d68c1-bb1a-4342-a5da-fd4843e42845.png)
When adding the code to fix it and entering a new window again, the error eliminates.
![image](https://user-images.githubusercontent.com/16870890/192798744-3c6be966-f8a0-4c21-8d95-ee42500e4f75.png)

But it didn't give any information about the number of instances over 80 categories.

**Comment by alice-cool** [comment]
![image](https://user-images.githubusercontent.com/16870890/192934724-b69b8687-c324-479e-9792-1bfcaafc0004.png)
When using multiple GPUs to run the code, the code throws the exception like the above picture...

Now I did the following modifications:
①in train_egine.py
![image](https://user-images.githubusercontent.com/16870890/192992897-c53f3d24-4c85-4d5b-aeca-eab4d983d535.png)

②in test_egine.py
Just narrow tab it. 
![image](https://user-images.githubusercontent.com/16870890/192993158-dc05fc67-ea38-41e4-9b8e-e473d586f1d7.png)

And the error eliminate. Because of the large scale input feature,  the code is running  for a long time....

**Comment by rentainhe** [comment]
> https://github.com/facebookresearch/grid-feats-vqa is it right?

Yes, you can use this code to extract 16x16 features, and we have also released our `feature-extraction code` here: [TRAR-Feature-Extraction](https://github.com/rentainhe/TRAR-Feature-Extraction), maybe you can extract 16x16 features by modifing the configs

**Comment by rentainhe** [comment]
> Dear scholar, I want to extract the 16 times 16 features using the ResNext152 described in your paper.

Maybe you can download it by using this scripts:
```bash
wget https://dl.fbaipublicfiles.com/grid-feats-vqa/X-152/X-152.pth
```

I borrowed it from here: https://github.com/facebookresearch/grid-feats-vqa#pre-trained-models-and-features

**Comment by alice-cool** [comment]
Thanks for&nbsp; your timely reply. I will try it now.



---原始邮件---
发件人: "Ren ***@***.***&gt;
发送时间: 2022年9月29日(星期四) 下午5:09
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [rentainhe/TRAR-VQA] --weights_path ./weight/X-152.pth wheredownload? (Issue #2)




  
https://github.com/facebookresearch/grid-feats-vqa is it right?
  
Yes, you can use this code to extract 16x16 features, and we have also released our feature-extraction code here: TRAR-Feature-Extraction, maybe you can extract 16x16 features by modifing the configs

**Comment by rentainhe** [comment]
> When I INPUT "CUDA_VISIBLE_DEVICES=1 python extract_trar_grid_feature.py --config-file configs/X-152-grid.yaml --dataset coco_2015_test --weight_path ./weight/X-152.pth --output_dir ./data/" to extract from coco test2015.zip jpg. It throws the exception like the following:
> 
> **raise KeyError( KeyError: "**Dataset 'coco_2015_test' is not registered!
> 
> ![image](https://user-images.githubusercontent.com/16870890/192797994-d97d68c1-bb1a-4342-a5da-fd4843e42845.png) When adding the code to fix it and entering a new window again, the error eliminates. ![image](https://user-images.githubusercontent.com/16870890/192798744-3c6be966-f8a0-4c21-8d95-ee42500e4f75.png)
> 
> But it didn't give any information about the number of instances over 80 categories.

I have no idea about this problem, cuz I think this may be specified settings in detectron2, I've try to extract GQA features using this code, however, it did not print the categories information too.

**Comment by alice-cool** [comment]
> > When I INPUT "CUDA_VISIBLE_DEVICES=1 python extract_trar_grid_feature.py --config-file configs/X-152-grid.yaml --dataset coco_2015_test --weight_path ./weight/X-152.pth --output_dir ./data/" to extract from coco test2015.zip jpg. It throws the exception like the following:
> > **raise KeyError( KeyError: "**Dataset 'coco_2015_test' is not registered!
> > ![image](https://user-images.githubusercontent.com/16870890/192797994-d97d68c1-bb1a-4342-a5da-fd4843e42845.png) When adding the code to fix it and entering a new window again, the error eliminates. ![image](https://user-images.githubusercontent.com/16870890/192798744-3c6be966-f8a0-4c21-8d95-ee42500e4f75.png)
> > But it didn't give any information about the number of instances over 80 categories.
> 
> I have no idea about this problem, cuz I think this may be specified settings in detectron2, I've try to extract GQA features using this code, however, it did not print the categories information too.

Thanks for your timely reply.

**Comment by rentainhe** [comment]
> ![image](https://user-images.githubusercontent.com/16870890/192934724-b69b8687-c324-479e-9792-1bfcaafc0004.png) When using multiple GPUs to run the code, the code throws the exception like the above picture...
> 
> Now I did the following modifications: ①in train_egine.py ![image](https://user-images.githubusercontent.com/16870890/192992897-c53f3d24-4c85-4d5b-aeca-eab4d983d535.png)
> 
> ②in test_egine.py Just narrow tab it. ![image](https://user-images.githubusercontent.com/16870890/192993158-dc05fc67-ea38-41e4-9b8e-e473d586f1d7.png)
> 
> And the error eliminate. Because of the large scale input feature, the code is running for a long time....

Thanks a lot! I've no idea if this is a bug in my code (cuz this code has not been updated for a long time), if this is a bug, would you like to submit a pull request to fix it ? : )

**Comment by erjpc** [comment]
> 

Hello, have you solved this problem? I want to extract 16*16, but it cannot be extracted correctly. I can't reproduce the 8*8 I extracted myself. Thank you for your help
